### DISCUSSION TITLE
How to close a sheet programatically?

### DISCUSSION THREAD

**Discussion by znat** [shadcn-ui/ui#1487] [Q&A] [answered]
I'd like to close the sheet when a query returns, the examples only shows the `<SheetClose />` tag
Thanks

**Comment by joaom00** [comment]
```tsx
const [open, setOpen] = React.useState(false)
<Sheet open={open} onOpenChange={setOpen}>
...
</Shet>
```

**Reply by znat** [reply]
Thanks, it works but it has the side effect that when state is explicitly controlled, closing with the cross is not possible anymore.
The icon is wrapped in a  `<SheetPrimitive.Close />` tag in `<SheetPrimitive.Content />` and there's no prop that can be leveraged to deal with that.
Do you have an idea on how to work around that?

[<SheetPrimitive.Close className="absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-secondary">](https://github.com/znat/nextboil/blob/471e3f3c7582cab042f6b6cbabc35c988525d929/components/ui/sheet.tsx#L74)

**Reply by joaom00** [reply, marked as answer]
I'm pretty sure that should work [Codesandbox](https://codesandbox.io/p/sandbox/hardcore-franklin-gvpx4d?file=/App.jsx:8,27)

**Reply by znat** [reply]
Oh right, I didn't understand `onOpenChange`at first. Thanks a lot for the codesandbox!
